commerce-cloud-code/app_yourshophere/cartridge/partials/order/orderitems.js

```diff
@@ -31,7 +31,7 @@ exports.createModel = function createModel(options) {
 
     const model = {
         empty: false,
-        orderNo: order.orderNo,
+        orderNo: options.settings.showOrderNumber !== false ? order.orderNo : '',
         orderDetailsTitle: options.settings.showOrderNumber !== false
```

Comment:
@SGD1953 As every good developer, I just copy and pasted your approach

`options.settings.showOrderNumber !== false`
You check for explicit false, because the default i.e null should be interpreted as null I suppose?

Maybe we can rather use an enum with ['show', 'hide'] instead of booleans? The JS tri-state booleans are a little confusing at times